scripts/west_commands/run_common.py

```diff
@@ -165,21 +192,72 @@ def do_run_common(command, user_args, user_runner_args, domains=None):
             # Get the user specified domains.
             domains = load_domains(build_dir).get_domains(user_args.domain)
 
-    if len(domains) > 1 and len(user_runner_args) > 0:
-        log.wrn("Specifying runner options for multiple domains is experimental.\n"
-                "If problems are experienced, please specify a single domain "
-                "using '--domain <domain>'")
+    if len(domains) > 1:
+        if len(user_runner_args) > 0:
+            log.wrn("Specifying runner options for multiple domains is experimental.\n"
+                    "If problems are experienced, please specify a single domain "
+                    "using '--domain <domain>'")
+
+        # Process all domains to load board names and populate flash runner
+        # parameters.
+        for d in domains:
+            if d.build_dir is None:
+                build_dir = get_build_dir(user_args)
+            else:
+                build_dir = d.build_dir
+
+            cache = load_cmake_cache(build_dir, user_args)
+            board = cache['CACHED_BOARD']
+
+            if board in board_image_count:
+                board_image_count[board][1] = board_image_count[board][1] + 1
+            else:
+                # This sets up the default entry for a board, no flashed images
+                # and one image in total.
+                board_image_count[board] = [0, 1]
```

Comment:
@mbolivar-nordic Would you want these changes too to have labels instead of just being an array (whatever the python type here is)?

Reply:
> Would you want these changed too, to have labels instead of just being an array (whatever the python type here is)?

@nordicjm just saw this one -- sounds good if you have time; you can used a dataclass here too, or a namedtuple.